sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierSerializerTest.cs

```diff
@@ -98,73 +99,82 @@ public void SerializeCallingApplication()
+            Assert.AreEqual(expectedIdentifier.Id, identifier.Id);
             Assert.AreEqual(expectedIdentifier, identifier);
         }
 
         [Test]
-        public void SerializePhoneNumber()
+        [TestCase(null)]
+        [TestCase("some id")]
+        public void SerializePhoneNumber(string? expectedId)
         {
-            CommunicationIdentifierModel model = CommunicationIdentifierSerializer.Serialize(new PhoneNumberIdentifier("+12223334444"));
+            CommunicationIdentifierModel model = CommunicationIdentifierSerializer.Serialize(new PhoneNumberIdentifier("+12223334444", expectedId));
 
             Assert.AreEqual(CommunicationIdentifierKind.PhoneNumber, model.Kind);
             Assert.AreEqual("+12223334444", model.PhoneNumber);
+            Assert.AreEqual(expectedId, model.Id);
         }
 
         [Test]
         public void DeserializePhoneNumber()
         {
-            CommunicationIdentifier identifier = CommunicationIdentifierSerializer.Deserialize(
+            PhoneNumberIdentifier identifier = (PhoneNumberIdentifier)CommunicationIdentifierSerializer.Deserialize(
                 new CommunicationIdentifierModel(CommunicationIdentifierKind.PhoneNumber)
                 {
                     PhoneNumber = "+12223334444",
+                    Id = "some id"
                 });
 
-            PhoneNumberIdentifier expectedIdentifier = new PhoneNumberIdentifier("+12223334444");
+            PhoneNumberIdentifier expectedIdentifier = new PhoneNumberIdentifier("+12223334444", "some id");
 
-            Assert.True(identifier is PhoneNumberIdentifier);
-            Assert.AreEqual(expectedIdentifier.PhoneNumber, ((PhoneNumberIdentifier)identifier).PhoneNumber);
+            Assert.AreEqual(expectedIdentifier.PhoneNumber, identifier.PhoneNumber);
+            Assert.AreEqual(expectedIdentifier.Id, identifier.Id);
             Assert.AreEqual(expectedIdentifier, identifier);
         }
 
         [Test]
-        [TestCase(false)]
-        [TestCase(true)]
-        public void SerializeMicrosoftTeamsUser(bool isAnonymous)
+        [TestCase(false, null)]
+        [TestCase(true, null)]
+        [TestCase(false, "some id")]
+        [TestCase(true, "some id")]
+        public void SerializeMicrosoftTeamsUser(bool isAnonymous, string? expectedId)
         {
-            CommunicationIdentifierModel model = CommunicationIdentifierSerializer.Serialize(new MicrosoftTeamsUserIdentifier("some id", isAnonymous));
+            CommunicationIdentifierModel model = CommunicationIdentifierSerializer.Serialize(new MicrosoftTeamsUserIdentifier("user id", isAnonymous, expectedId));
 
             Assert.AreEqual(CommunicationIdentifierKind.MicrosoftTeamsUser, model.Kind);
-            Assert.AreEqual("some id", model.MicrosoftTeamsUserId);
+            Assert.AreEqual("user id", model.MicrosoftTeamsUserId);
             Assert.AreEqual(isAnonymous, model.IsAnonymous);
+            Assert.AreEqual(expectedId, model.Id);
```

Comment:
Let's add an Assert that Cloud is set to Public when it's missing. Alos that IsAnonymous is set to false when it's missing in the constructor call.

Reply:
I have a test in the identifier types for this as in C# we do it during construction.